gate-web/src/main/groovy/com/netflix/spinnaker/gate/filters/FiatSessionFilter.groovy

```diff
@@ -53,10 +54,13 @@ class FiatSessionFilter implements Filter {
       String user = AuthenticatedRequest.getSpinnakerUser().orElse(null)
       if (permissionEvaluator.getPermission(user) == null) {
         HttpServletRequest httpReq = (HttpServletRequest) request
-        log.info("Invalidating user '${user}' session '${httpReq.session.id}'" +
-                     " because Fiat permission was not found.")
-        httpReq.session.invalidate()
-        SecurityContextHolder.clearContext()
+        HttpSession session = httpReq.getSession(false)
```

Comment:
@ttomsu small tweak here to avoid creating sessions when the request doesn't already have one